src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,85 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private static Map<Class<?>, Class<?>> implementationMap;
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
+        implementationMap = fillDefaultStorage();
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (!interfaceClazz.isInterface()
+                && interfaceClazz.getDeclaredAnnotation(Component.class) == null) {
+            throw new RuntimeException(
+                    "Injection failed, missing @Component annotation on the class "
+                    + interfaceClazz.getName());
+        }
+
+        Object clazzImplementationInstance = null;
+        Class<?> implementation = findImplementation(interfaceClazz);
+        Field[] fields = implementation.getDeclaredFields();
+        for (Field field : fields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+
+                clazzImplementationInstance = createNewInstance(implementation);
+
+                try {
+                    field.setAccessible(true);
+                    field.set(clazzImplementationInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Can`t modify field "
+                            + field.getName() + " in " + interfaceClazz.getName(), e);
+                }
+            }
+        }
+        if (clazzImplementationInstance == null) {
+            clazzImplementationInstance = createNewInstance(implementation);
+        }
+        return clazzImplementationInstance;
+    }
+
+    private Object createNewInstance(Class<?> clazz) {
+        if (instances.containsKey(clazz)) {
+            return instances.get(clazz);
+        }
+        Object instance;
+        try {
+            instance = clazz.getConstructor().newInstance();
+            instances.put(clazz, instance);
+        } catch (InstantiationException | IllegalAccessException
+                 | InvocationTargetException | NoSuchMethodException e) {
```

Comment:
You can change 
```suggestion
        } catch (ReflectiveOperationException e) {
```
That was in common mistakes